src/osd.c

```diff
@@ -204,6 +204,7 @@ osd_update(struct server *server)
 			pango_cairo_show_layout(cairo, layout);
 			y += OSD_ITEM_HEIGHT;
 		}
+		free(buf.buf);
```

Comment:
`buf_init()` is called with the `wl_list_for_each(..., &server->outputs, ...)` which isn't great.
We ought to move these two lines to before the foreach loop

```
struct buf buf;
buf_init(&buf);
```

And then in the loop just set buf.len = 0 to clear it.

That way we know buf.buf can be freed. If we have not outputs we risk freeing a NULL pointer otherwise.

Reply:
Isn't the new `free()` within the same loop as well though?

> We ought to move these two lines to before the foreach loop
> And then in the loop just set buf.len = 0 to clear it.

Yes, that is a good idea, will change